src/main/java/org/wso2/carbon/esb/connector/utils/SalesforceUtils.java

```diff
@@ -250,7 +260,60 @@ public static void generateJsonOutput(MessageContext messageContext, String json
         }
     }
 
-    public static String csvToJson(String csvString) {
+    public static String csvToJson(String csvString) throws IOException {
+
+        String escapedString = StringEscapeUtils.unescapeXml(csvString);
+        CSVReader csvReader =
+                new CSVReader(new StringReader(escapedString), CSVReader.DEFAULT_SEPARATOR, CSVReader.DEFAULT_QUOTE_CHARACTER);
+        List<String[]> lines = csvReader.readAll();
+        String[] headerRow = lines.get(0);
+
+        for (int i = 0; i < headerRow.length; i++) {
+            headerRow[i] = removeQuotes(headerRow[i]);
+        }
+        Stream<String[]> csvArrayStream = lines.stream();
+        csvArrayStream = csvArrayStream.skip(1);
+        List<JsonObject> jsonObjectList = csvArrayStream
+                .map(row -> {
+                    JsonObject jsonObject = new JsonObject();
+
+                    for (int i = 0; i < row.length; i++) {
+                        JsonPrimitive value = getCellValue(row, i);
+                        String key = headerRow[i];
+                        jsonObject.add(key, value);
+                    }
+
+                    return jsonObject;
+                })
+                .collect(Collectors.toList());
+        Gson gson = new GsonBuilder()
+                .disableHtmlEscaping()
+                .setPrettyPrinting()
+                .create();
+        String jsonString = gson.toJson(jsonObjectList);
+        jsonString = jsonString.replace("\\\\", "\\");
```

Comment:
Can you please explain the purpose of this         jsonString = jsonString.replace("\\\\", "\\");

Reply:
Earlier, it was escaping the escape characters again and transformed '\n' to '\\n' like these. It is not needed anymore. I'll remove it.